### PR TITLE
Added freenome-build db options for databases in Kubernetes

### PR DESCRIPTION
**Pull request by ianlo**
Added freenome-build db start-k8s, start-k8s-test-db, and stop-k8s

This assumes that any kubernetes test db we create will want to run on port 5432 of its pod.

I'm a little torn on whether or not it would be better to create a k8s-db subparser especially since the insert and reset commands remain the same. Let me know your thoughts!

### CODE REVIEW

File: freenome_build/db.py [hunk 1]
```diff
@@ -156,6 +193,28 @@ def start_local_database(repo_path: str, project_name: str, port: int = None, pa
     return conn_data
 
 
+def start_k8s_database(repo_path: str, project_name: str, password: str = None,
+                       kube_pod_config: str = None) -> Tuple[DbConnectionData, str]:
+    dbname = project_name
```
Comment (nboley) [inline freenome_build/db.py]: I would make dbname and user be arguments, and let the caller decide to use project_name for both.

File: freenome_build/db.py [hunk 2]
```diff
@@ -156,6 +193,28 @@ def start_local_database(repo_path: str, project_name: str, port: int = None, pa
...
+    user = project_name
+    port = 5432
```
Comment (nboley) [inline freenome_build/db.py]: might as well make this an argument with 5432 as the default.

File: freenome_build/db.py [hunk 3]
```diff
@@ -156,6 +193,28 @@ def start_local_database(repo_path: str, project_name: str, port: int = None, pa
...
+
+    if kube_pod_config is None:
+        kube_pod_config = norm_abs_join_path(
+            os.path.dirname(__file__), "./database_template/db_pod_config.yaml")
```
Comment (nboley) [inline freenome_build/db.py]: Make sure to update the README with a description.

File: freenome_build/db.py [hunk 4]
```diff
@@ -254,12 +313,27 @@ def stop_local_database(conn_data: DbConnectionData) -> None:
...
 
 
+def start_k8s_database_main(args):
+    if args.conn_data:
+        conn_data, pod_id = start_k8s_database(args.path, args.conn_data.dbname,
+                                               port=args.conn_data.port, kube_pod_config=args.kube_pod_config)
+    else:
+        conn_data, pod_id = start_k8s_database(args.path, args.project_name, kube_pod_config=args.kube_pod_config)
+    logger.info(f"Successfully started a database. Connect to {pod_id} and use the following string to connect:")
+    print(conn_data)
```
Comment (nboley) [inline freenome_build/db.py]: logger?

File: freenome_build/db.py [hunk 5]
```diff
@@ -296,17 +386,29 @@ def stop_local_database_main(args):
...
+
+
 def add_db_subparser(subparsers):
     database_parser = subparsers.add_parser('db', help='manage the test database')
     database_parser.required = True
     database_parser.add_argument('--path', default='.')
     database_parser.add_argument(
-        '--port', type=int,
-        help='Port on which to start the test db. Default is a random free port'
+        '--host', type=str,
```
Comment (nboley) [inline freenome_build/db.py]: Why did you change the description and remove the `localhost` default?

File: tests/test_test_db.py [hunk 6]
```diff
@@ -67,3 +69,50 @@ def test_db_module_interface():
     insert_test_data(conn_data, DB_DIR)
     reset_data(conn_data, DB_DIR)
     stop_local_database(conn_data)
+
+
+def _test_k8s_connection(testing_pod_id: str, conn_data: DbConnectionData):
```
Comment (nboley) [inline tests/test_test_db.py]: nice testing!

File: freenome_build/db.py (hunk 4 above)
Comment (ianlo) [reply freenome_build/db.py]: We should keep this as print so that we can give the user the connection string in stdout right?

File: freenome_build/db.py (hunk 5 above)
Comment (ianlo) [reply freenome_build/db.py]: Good point. I'll switch this back

File: freenome_build/db.py (hunk 4 above)
Comment (nboley) [reply freenome_build/db.py]: ah, that makes sense. Maybe add a comment.
